libkbfs/md_ops.go

```diff
@@ -227,6 +327,30 @@ func (md *MDOpsStandard) verifyKey(
 		// the server could be lying to us.
 		md.log.CDebugf(ctx, "The server claims there haven't been any "+
 			"KBFS merkle trees published since the revocation")
+
+		// Check the most recent global merkle root and KBFS merkle
+		// root ctimes and make sure they fall within the expected
+		// error window with respect to the revocation.
+		_, latestRootTime, err := md.config.KBPKI().GetCurrentMerkleRoot(ctx)
+		if err != nil {
+			return false, err
+		}
+		treeID := mdToMerkleTreeID(irmd)
+		// TODO: cache the latest KBFS merkle root somewhere for a while?
+		latestKbfsRoot, err := md.config.MDServer().GetMerkleRootLatest(
+			ctx, treeID)
+		if err != nil {
+			return false, err
+		}
+		serverOffset, _ := md.config.MDServer().OffsetFromServerTime()
+		currServerTime := md.config.Clock().Now().Add(-serverOffset)
+		err = md.checkMerkleTimes(
+			ctx, latestRootTime, latestKbfsRoot, currServerTime,
```

Comment:
`currServerTime` here is basically server-trust right? I wonder if there's a way to use a NTP time, or perhaps we should just use local time. It's probably reasonable to assume the local time is roughly in sync with a NTP time. If the local time is terribly out-of-sync (e.g. dead motherboard battery and no NTP service running), HTTPS would be broken too.

Reply:
As you know I'm very reluctant to use local time for anything.  I have seen clocks be out of sync too many times, even server clocks run by companies.

You're right though, that this particular instance of using server trust is a bit problematic.  Luckily it should be pretty rare, since it should only happen when a client is reading an MD update within ~3 hours of the device being revoked.

How about I just add a check here to make sure that `abs(serverOffset)` is less than 1 hour?  If that breaks, than the worst that can happen is the client will have to wait for the next merkle run.

Reply:
> How about I just add a check here to make sure that abs(serverOffset) is less than 1 hour?

That sounds good with me. Thanks!